soccer/gameplay/situational_play_selection.py

```diff
@@ -298,60 +318,64 @@ def isMidfield(self):
     def isPileup(self):
         return self.currentPileup
 
-    ##Returns if we are in the specified situation without regard to capitalization
-    #If the check variable is true, it will check if the situation exists and throw an
-    #exception is it does now.
-    def isSituation(self, situation, check=False):
-        up = situation.upper()
-
-        if (check):
-            found = False
-            for g in self.Situation:
-                if (up == g):
-                    found = True
-                    break
-            if not found:
-                raise Exception("Passed situation " + situation + " / " + up +
-                                " is not an existing situation")
-
-        if (self.currentSituaion.name == up):
-            return True
+    ##Returns the current situation
+    def getSituation(self):
+        return self.currentSituation
+
+    ##
+    # Returns if we are in the passed situation
+    # 
+    # @param situation a situation as an enum
+    def isSituation(self, situation):
+        if (isinstance(situation, self.Situation)):
+            return situation == self.currentSituation
         else:
-            return False
+            raise TypeError("isSituation only takes enums")
 
-    ##Update determining if we want to preempt the current play or not
-    #
-    # Preemption is still an open question but this is a prototype of
-    # of how a non-invasive preemption system might work
-    #
-    def updatePreempt(self):
-        if (self.lastSituation != self.currentSituaion and
-                self.situationChangeTime == None):
-            self.situationChangeTime = time.time()
+    ##Returns true if we are in any of the situations in the passed list
+    #passed situation list must be enums
+    def isSituations(self, situations):
+        return self.currentSituation in situations
 
-        if (self.currentPreempt):
-            self.currentPreempt = False
-        elif (self.situationChangeTime != None):
-            if (abs(time.time() - self.situationChangeTime) >
-                    self.playPreemptTime):
-                self.situationChangeTime = None
-                self.currentPreempt = True
-                self.LastSituation = self.currentSituaion
+    ##
+    # 
+    # Will determine if the current play has lasted too long over a situation change
+    #   and needs to be preempted, it will call try_preempt
+    def updatePreempt(self):
```

Comment:
Correct me if I'm wrong, but this will preempt on every situation change even if the current play accounts for this situation? Is that the behavior we want? I personally would let the current play continue to run so we don't loose state info from just restarting it.

Reply:
So I think that on one hand if we are running in a gimped situation where we have one play covering many adjacent situations and are using our current plays, we will get a performance benefit from not preempting switches from and to the same play. But I think at this point it's more valuable to preempt them and see how everything handles it since that will be the expectation when we have good play coverage, our plays should be able to handle being started and stopped because that is how it's going to be with more plays in the playbook. For that reason I have kept it preempting switches between the same plays.